application/src/main/java/org/togetherjava/tjbot/features/reminder/RemindRoutine.java

```diff
@@ -142,4 +146,35 @@ static ReminderRoute toPrivate(PrivateChannel channel) {
                             + " the original channel you wanted it to be send to)");
         }
     }
-}
+
+    private void reminderFailureAction(int id, long channelId, long authorId,
+                                       CharSequence content, long guildId, int reminderFailureCounter) {
+        if (reminderFailureCounter <= 3) {
+            Instant current = Instant.now();
+            database.write(context -> context.newRecord(PENDING_REMINDERS)
+                    .setId(id)
+                    .setCreatedAt(Instant.now())
+                    .setGuildId(guildId)
+                    .setChannelId(channelId)
+                    .setAuthorId(authorId)
+                    .setRemindAt(current.plusSeconds(60))
+                    .setContent(content.toString())
+                    .setReminderFailureCounter(reminderFailureCounter + 1)
+                    .insert());
+        } else {
+            logger.warn(
+                    """
+                            Failed to send a reminder with (authorID '{}'), skipping it. This can be due to a network issue, \
+                            but also happen if the bot disconnected from the target guild and the \
+                            user has disabled DMs or has been deleted."""+ "contents: "+content,
+                    authorId);
+        }
+    }
+
+        private Integer getReminderFailureCounter(int id){
+            return database.read(context -> context.selectFrom(PENDING_REMINDERS).where(PENDING_REMINDERS.ID.eq(id))
+                    .stream().map(PendingRemindersRecord::getReminderFailureCounter).findFirst().get()
+            );
+        }
```

Comment:
it is unecessary that u read the database again. u can get hands on the failure counter where u read it originally, here:

![original](https://i.imgur.com/dYCoKSL.png)

Reply:
fixed it